### ISSUE TITLE
Allow autogenerated names back into in the workspace URL

### ISSUE THREAD

**Issue by apolopena** [gitpod-io/gitpod#7737]
## Is your feature request related to a problem? Please describe
**Context:**
This pertains to a typical workflow where the user is running multiple instances of the same workspace. I do this often when testing out and comparing gitpod configurations.

**The Problem:**
Now that the autogenerated name (i.e.
harlequin-crayfish) in the workspace URL has been replaced with the repository name there is no real good way to identify multiple versions of the same workspace running in gitpod. 

Two workspace urls open side by side will look something like this:
https://gitpodio-apacheexample-xpbkletuqfe.ws-us27.gitpod.io/
https://gitpodio-apacheexample-tmbletixgk.ws-us27.gitpod.io/

To make things even harder, now that the repo name is in the first part of the workspace URL two tabs in the browser can look exactly the same:
![image](https://user-images.githubusercontent.com/3060702/150418415-998080ec-70d6-4caf-bcbf-5ee502b42ada.png)

## Describe the behaviour you'd like

There are Many ways to handle this. One flexible solution would be to have a setting in the dashboard that allows the user to toggle the ability to prefix the workspace url (as it was before) with a somewhat unique human readable autogenerated name (i.e.
harlequin-crayfish) either in addition to the repository name or replacing it.

## Describe alternatives you've considered

I can't think of any workarounds since the generation of the workspace URL is not exposed to the user.

## Additional context

Putting the repository name in the workspace url is generally helpful and I cant totally see why this was done but it does affect the workflow where a user runs mulitple versions of the same workspace in that if becomes almost impossible to distinguish them at a glance.

**Comment by ashooner** [comment]
Seconding this. It seems like a regression to make users distinguish hashes at the end of otherwise identical repo-based string when the previous unique names addressed this exact problem.

**Comment by pawlean** [comment]
@shaal reminded me that you can also rename workspaces in your workspace dashboard: https://gitpod.io/workspaces 🙏🏼

![2022-01-21 at 18 25 52 - Google Chrome](https://user-images.githubusercontent.com/17087373/150580500-f8e5a077-6ec0-4580-b4b3-7aca5346b435.png)

**Comment by pawlean** [comment]
Another thing that @shaal shared on the [Discord discussion](https://discord.com/channels/816244985187008514/816246578594840586/934097175279988747) which may be helpful:

> @Pauline 🍊💜 Perhaps a missing improvement would be updating the document.title ?
> For example, on a current project I am working on, I updated the workspace name to HF-863 - Diffy setup
> (the Jira ticket - and a friendly name of the purpose for the branch)
> But what the browser is showing on the open tab is (document.title) -
> 'composer.json — hf-drupal — Gitpod Code'
> It's the current open file, name of repo, 'Gitpod Code'

**Comment by apolopena** [comment]
> Another thing that @shaal shared on the [Discord discussion](https://discord.com/channels/816244985187008514/816246578594840586/934097175279988747) which may be helpful:
> 
> > @pauline 🍊💜 Perhaps a missing improvement would be updating the document.title ?
> > For example, on a current project I am working on, I updated the workspace name to HF-863 - Diffy setup
> > (the Jira ticket - and a friendly name of the purpose for the branch)
> > But what the browser is showing on the open tab is (document.title) -
> > 'composer.json — hf-drupal — Gitpod Code'
> > It's the current open file, name of repo, 'Gitpod Code'

Yeah if this feature is implemented the browser title should start with the autogenerated name, otherwise the feature only solves half the problem.

**Comment by shaal** [comment]
Please add `never-stale`  label

**Comment by axonasif** [comment]
<img width="886" alt="Screenshot 2022-06-30 at 9 52 30 PM" src="https://user-images.githubusercontent.com/39482679/176722619-20045b38-6a98-482c-b4e6-4a1d53c9d7c2.png">

Source: https://discord.com/channels/816244985187008514/902832281369075722/992085973561245697

**Comment by david-bakin** [comment]
And related #10773 about visibility of (manually assigned) workspace names.
